### ISSUE TITLE
Lerna publish with pnpm workspace protocol is not updating package version when it is a dependency

### ISSUE THREAD

**Issue by batusai513** [lerna/lerna#3456]
## Current Behavior

When I `lerna publish`, the dependencies that use the workspace protocol as version are not updated to the latest version in the generated `package.json` artifact uploaded to npm.

## Expected Behavior

When I run `lerna publish` all packages affected which are dependencies of other packages should also be updated.

### before lerna publish
```json
{
  "name": "package-1",
  "version": "1.0.0"
}
```
```json
{
  "name": "package-2",
  "version": "2.0.0",
  "dependencies": {
    "package-1": "workspace:1.0.0"
  }
}
```

### after lerna publish in the repository code
```json
{
  "name": "package-1",
  "version": "1.1.0"
}
```
```json
{
  "name": "package-2",
  "version": "2.1.0",
  "dependencies": {
    "package-1": "workspace:1.1.0"
  }
}
```
But when installing the updated package from npm and inspection it's `package.json` file inside `node_modules` folder
```json
{
  "name": "package-2",
  "version": "2.1.0",
  "dependencies": {
    "package-1": "1.0.0" //previous version
  }
}
```

### Failure Logs / Configuration

<details><summary>lerna.json</summary><p>

```json
{
  "$schema": "node_modules/lerna/schemas/lerna-schema.json",
  "packages": ["packages/*"],
  "version": "independent",
  "npmClient": "pnpm",
  "useWorkspaces": true,
  "changelogPreset": "conventionalcommits",
  "command": {
    "publish": {},
    "version": {
      "allowBranch": ["master"],
      "conventionalCommits": true,
      "createRelease": "github",
      "exact": true,
      "message": "chore(release): publish"
    }
  }
}
```

</p></details>

<details><summary>lerna-debug.log</summary><p>

</p></details>

**Comment by fahslaj** [comment]
Hi @batusai513 Thank you for reporting this issue. Unfortunately, I was unable to reproduce this issue using the latest version of Lerna with the information that you have provided. In order for us to troubleshoot this issue, please kindly provide a minimal reproduction of the issue. You can use https://github.com/lerna/repro as a starting point.

**Comment by candy-Tong** [comment]
I also has similar problems.

It is ok when i release the package after write some code.
But I release again after release and do not add any code, it will get a error because i do not add any code and the verson of all the packages are not change. So it push fail.
```
lerna ERR! E403 You cannot publish over the previously published versions: 0.5.9.
 ELIFECYCLE  Command failed with exit code 1.
```

At this time, the verson in package.json will be change like this.
![image](https://user-images.githubusercontent.com/20377280/211482248-d8578107-d397-4472-aee5-1be3450f3826.png)

**Comment by Leksat** [comment]
I face the same issue as described in the first comment.

- `@amazeelabs/publisher` depends on `@amazeelabs/publisher-shared`
- I updated both packages ([commit](https://github.com/AmazeeLabs/silverback-mono/commit/fd17f87e762a106dc3d68995e108a44061f38411))
- the following `lerna publish` ([commit](https://github.com/AmazeeLabs/silverback-mono/commit/3a8bc4a15d25afb4ddb625fe74165aad83a39a1a))
  - updated version of both packages from `2.0.1` to `2.0.2`
  - updated the dependency in `@amazeelabs/publisher` as follows
    ```
    "@amazeelabs/publisher-elements": "workspace:^2.0.2"
    ```
    ([change](https://github.com/AmazeeLabs/silverback-mono/commit/3a8bc4a15d25afb4ddb625fe74165aad83a39a1a#diff-c98c85fb54078c7e0b14bf403569b8ff03b4681d8d564a0747c7df1ef3adf975))

Everything seems perfectly fine, but in the version published on NPM, the dependency version is still `^2.0.1`: https://www.runpkg.com/?@amazeelabs/publisher@2.0.2/package.json#22

**Comment by Leksat** [comment]
@fahslaj If I create a repro, should it publish to NPM? Or is there a way to tell lerna to keep the packed version locally rather than upload it to NPM?

**Comment by fahslaj** [comment]
@Leksat You could use [Verdaccio](https://verdaccio.org/) to host a private npm registry locally to test your repro. You would set your npm registry to the verdaccio url (something like http://localhost:4873), perform the publish, then navigate to that verdaccio url to check the published assets.

**Comment by Leksat** [comment]
@fahslaj https://github.com/lerna/repro/pull/10 I hope it helps :)

**Comment by fahslaj** [comment]
Thank you for the reproduction @Leksat ! I see the issue and will have a fix soon.

**Comment by fahslaj** [comment]
The fix for this has been released in Lerna 6.6.1 🎉
